### PR TITLE
Improve automation benchmark test

### PR DESCRIPTION
**Pull request by anirudhwarrier**
- support multi registry tests
- add test feature - ForceSingleTxnKey, DeleteJobsOnEnd
- reuse link, feed contracts for multi registry test
- add test for testing all different registry version on same nodes

### CODE REVIEW

File: integration-tests/benchmark/keeper_test.go [hunk 1]
```diff
@@ -195,12 +203,15 @@ func TestAutomationBenchmark(t *testing.T) {
...
 			ChainlinkNodeFunding:  benchmarkTestEntry.funding,
-			UpkeepGasLimit:        PerformGasToBurn + 50000,
+			UpkeepGasLimit:        UpkeepGasLimit,
 			UpkeepSLA:             benchmarkTestEntry.upkeepSLA,
 			FirstEligibleBuffer:   1,
 			PreDeployedConsumers:  benchmarkTestEntry.predeployedConsumers,
 			UpkeepResetterAddress: benchmarkTestEntry.upkeepResetterAddress,
 			BlockTime:             benchmarkTestEntry.blockTime,
+			DeltaStage:            benchmarkTestEntry.deltaStage,
+			ForceSingleTxnKey:     false,
```
Comment (anirudhwarrier) [inline integration-tests/benchmark/keeper_test.go:213]:
```suggestion
			ForceSingleTxnKey:     benchmarkTestEntry. forceSingleTxnKey,
```

File: integration-tests/actions/automation_ocr_helpers.go [hunk 2]
```diff
@@ -31,36 +31,37 @@ func BuildAutoOCR2ConfigVars(
...
 		deltaStage,            // deltaStage time.Duration,
-		48,                    // rMax uint8,
+		24,                    // rMax uint8,
 		S,                     // s []int,
 		oracleIdentities,      // oracles []OracleIdentityExtra,
 		types2.OffchainConfig{
 			TargetProbability:    "0.999",
 			TargetInRounds:       1,
-			PerformLockoutWindow: 100 * 12 * 1000, // ~100 block lockout (on goerli)
+			PerformLockoutWindow: 3600000,
```
Comment (infiloop2) [inline integration-tests/actions/automation_ocr_helpers.go:51]: nit. let's add a comment here that this is intentionally different than prod

File: integration-tests/benchmark/keeper_test.go [hunk 3]
```diff
@@ -324,14 +393,64 @@ var tests = map[string]*BenchmarkTestEntry{
...
+		int64(10),
+		predeployedConsumersMumbai,
+		upkeepResetterContractMumbai,
+		3 * time.Second,
+		time.Duration(0),
+		false,
+		true,
+	},
+	"ArbitrumGoerliRegistry_2_0-X": {
+		repeatRegistries(eth_contracts.RegistryVersion_2_0, NumberOfRegistries),
```
Comment (infiloop2) [inline integration-tests/benchmark/keeper_test.go:423]: nit. can we rename this variable to 2_0_2 ?

File: integration-tests/testsetups/keeper_benchmark.go [hunk 4]
```diff
@@ -177,27 +193,43 @@ func (k *KeeperBenchmarkTest) Run(t *testing.T) {
...
-		// Reset upkeeps so that they become eligible gradually after the test starts
-		actions.ResetUpkeeps(t, contractDeployer, k.chainClient, inputs.NumberOfContracts, inputs.BlockRange, inputs.BlockInterval, inputs.CheckGasToBurn,
-			inputs.PerformGasToBurn, inputs.FirstEligibleBuffer, k.keeperConsumerContracts[rIndex], inputs.UpkeepResetterAddress)
+		for _, consumerContract := range k.keeperConsumerContracts[rIndex] {
+			consumerContractsToReset = append(consumerContractsToReset, consumerContract)
+			log.Info().Str("Contract", consumerContract.Address()).Msg("Added consumer contract to reset queue")
+		}
+	}
+
+	log.Info().Int("consumerContractsToReset", len(consumerContractsToReset)).Msg("Reset Queue Length")
```
Comment (anirudhwarrier) [inline integration-tests/testsetups/keeper_benchmark.go:227]:
```suggestion
	log.Debug().Int("consumerContractsToReset", len(consumerContractsToReset)).Msg("Reset Queue Length")
```

File: integration-tests/benchmark/keeper_test.go [hunk 5]
```diff
@@ -195,12 +203,15 @@ func TestAutomationBenchmark(t *testing.T) {
...
-			UpkeepGasLimit:        PerformGasToBurn + 50000,
+			UpkeepGasLimit:        UpkeepGasLimit,
 			UpkeepSLA:             benchmarkTestEntry.upkeepSLA,
 			FirstEligibleBuffer:   1,
 			PreDeployedConsumers:  benchmarkTestEntry.predeployedConsumers,
 			UpkeepResetterAddress: benchmarkTestEntry.upkeepResetterAddress,
 			BlockTime:             benchmarkTestEntry.blockTime,
+			DeltaStage:            benchmarkTestEntry.deltaStage,
+			ForceSingleTxnKey:     false,
+			DeleteJobsOnEnd:       false,
```
Comment (anirudhwarrier) [inline integration-tests/benchmark/keeper_test.go:214]:
```suggestion
			DeleteJobsOnEnd:       benchmarkTestEntry.deleteJobsOnEnd,
```

File: integration-tests/benchmark/keeper_test.go [hunk 6]
```diff
@@ -324,14 +393,64 @@ var tests = map[string]*BenchmarkTestEntry{
...
+		big.NewFloat(ChainlinkNodeFunding),
+		int64(10),
+		predeployedConsumersMumbai,
+		upkeepResetterContractMumbai,
+		3 * time.Second,
+		time.Duration(0),
+		false,
+		true,
+	},
+	"ArbitrumGoerliRegistry_2_0-X": {
```
Comment (anirudhwarrier) [inline integration-tests/benchmark/keeper_test.go:422]:
```suggestion
	"ArbitrumGoerliRegistry_2_0-Multiple": {
```

File: integration-tests/benchmark/keeper_test.go [hunk 7]
```diff
@@ -324,14 +393,64 @@ var tests = map[string]*BenchmarkTestEntry{
...
+		big.NewFloat(ChainlinkNodeFunding),
+		int64(10),
+		predeployedConsumersEmpty,
+		upkeepResetterContractEmpty,
+		1 * time.Second,
+		time.Second * 30,
+		true,
+		false,
+	},
+	"SimulatedGethRegistry_2_0-X": {
```
Comment (anirudhwarrier) [inline integration-tests/benchmark/keeper_test.go:433]:
```suggestion
	"SimulatedGethRegistry_2_0-Multiple": {
```

File: integration-tests/actions/ocr_helpers.go [hunk 8]
```diff
@@ -489,6 +489,10 @@ func BuildGeneralOCR2Config(
 }
 
 func getOracleIdentities(t *testing.T, chainlinkNodes []*client.Chainlink) ([]int, []confighelper.OracleIdentityExtra) {
+	return getOracleIdentitiesWithKeyIndex(t, chainlinkNodes, 0)
+}
+
+func getOracleIdentitiesWithKeyIndex(t *testing.T, chainlinkNodes []*client.Chainlink, keyIndex int) ([]int, []confighelper.OracleIdentityExtra) {
```
Comment (infiloop2) [inline integration-tests/actions/ocr_helpers.go:495]: 👍

**Pull request merged**